crates/ruff_python_formatter/src/comments/format.rs

```diff
@@ -151,18 +151,43 @@ impl Format<PyFormatContext<'_>> for FormatTrailingComments<'_> {
                 write!(
                     f,
                     [
-                        line_suffix(&format_args![
-                            empty_lines(lines_before_comment),
-                            format_comment(trailing)
-                        ]),
+                        line_suffix(
+                            &format_args![
+                                empty_lines(lines_before_comment),
+                                format_comment(trailing)
+                            ],
+                            0
+                        ),
                         expand_parent()
                     ]
                 )?;
             } else {
+                // TODO(cnpryer): Reserved width should be used when breaking is expected. Breaking may not be
+                //   expected when a trailing end of line comment is not at the end of a collapsed node.
+                // ```python
+                // def fn():
+                //     return np.divide(
+                //         view.variance,
+                //         view.sum_of_weights,
+                //         out=np.full(view.sum_of_weights.shape, np.nan),
+                //         where=view.sum_of_weights**2 > view.sum_of_weights_squared,  # This shouldn't force a break
+                //     )
+                // ```
```

Comment:
I believe I will need to address this

Reply:
The behavior you have is fine. We want to respect the line width for all comments, but we'll exclude trailing pragma comments in a follow up PR

Reply:
Removed the comment in 9385e63a278f5e0fbd8f6d78f15605d92e6ba18c

Reply:
Idk how I missed this, but this does cause it to break

```py
def fn():
    return np.divide(
        view.variance,
        view.sum_of_weights,
        out=np.full(view.sum_of_weights.shape, np.nan),
        where=view.sum_of_weights**2 > view.sum_of_weights_squared,  # This shouldn't force a break
        )
```
Becomes
```py
def fn():
    return np.divide(
        view.variance,
        view.sum_of_weights,
        out=np.full(view.sum_of_weights.shape, np.nan),
        where=view.sum_of_weights**2
        > view.sum_of_weights_squared,  # This shouldn't force a break
    )
```
With `black`